.github/workflows/trivy.yml

```diff
@@ -0,0 +1,50 @@
+name: Trivy Check CI
+on:
+  workflow_call: {}
+
+permissions:
+  contents: write
+
+env:
+  CI: 1
+
+jobs:
+  trivy-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+      - name: "Extract NIX_PATH"
+        id: nixpath
+        run: |
+          echo "nix_path=nixpkgs=$(jq -r '.nixpkgs.url' nix/sources.json)" >> $GITHUB_OUTPUT
+      - uses: cachix/install-nix-action@v31
+        with:
+          nix_path: "${{ steps.nixpath.outputs.nix_path }}"
+          install_options: --no-daemon
+          extra_nix_config: |
+            keep-env-derivations = true
+            keep-outputs = true
+          enable_kvm: true
+      - name: Restore and save Nix store
+        uses: nix-community/cache-nix-action@v6
+        with:
+          primary-key: nix-${{ hashFiles('shell.nix', 'nix/*') }}
+          restore-prefixes-first-match: nix-
+      - run: nix-build shell.nix
+      - uses: docker/setup-qemu-action@v3
+      - uses: docker/setup-buildx-action@v3
+        with:
+          driver-opts: |
+            network=host
+      - uses: aquasecurity/setup-trivy@v0.2.4
+      - name: Build and scan image
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          nix-shell --run ".ci/trivy.sh"
+      - name: Upload Trivy scan results to GitHub Security tab
```

Comment:
What happens on a PullRequest, where is report uploaded to?
Also how to scan existing releases?

Reply:
IDK about previous releases, but for PR it will show the results by a filter of PR id in GitHub Security scan section, checkout the example link that is commented automatically

Reply:
@tiagolobocastro

If that's Ok, I will merge

Reply:
Yes please do, was only a question